src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java

```diff
@@ -169,12 +172,46 @@ private void handle(String consumeTag, Delivery delivery) {
         }
     }
 
+    private boolean resubscribe() {
+        LOGGER.info("Trying to resubscribe to queue.");
+
+        RabbitMQConfiguration configuration = this.connectionManager.get().getConfiguration();
+        int attemptsCount = 0;
+        int amountOfAttempts = configuration.getMaxRecoveryAttempts();
+        boolean successful = false;
+
+        while (attemptsCount < amountOfAttempts) {
+            try {
+                Thread.sleep(configuration.getMaxConnectionRecoveryTimeout());
+            } catch (InterruptedException | IllegalArgumentException e) {
+                LOGGER.error(e.getMessage());
+            }
+
+            try {
+                attemptsCount++;
+                consumerMonitor.set(null);
+                start();
+
+                CommonMetrics.setReadiness(true);
+                successful = true;
+                break;
+            } catch (Exception e) {
+                if(attemptsCount >= amountOfAttempts) {
+                    LOGGER.error("Failed to start listening when resubscribing after {} attempts.", amountOfAttempts);
+                }
+            }
+        }
+
+        return successful;
+    }
+
     private void canceled(String consumerTag) {
```

Comment:
Are you sure that the rabbit library will not be broken if we block its tread to unsubscribe?
Please read about this case and check it.